docs/ngmodule.md

```diff
@@ -0,0 +1,206 @@
+# Keycloak Angular Integration with NgModule Application
+
+[Return to README.md](../README.md)
+
+---
+
+- [About](#about)
+- [Setup](#setup)
+- [Example project](#example-project)
+- [AuthGuard](#authguard)
+- [HttpClient Interceptor](#httpclient-interceptor)
+- [Keycloak-js Events](#keycloak-js-events)
+
+---
+
+## About
+
+This library helps you to use [keycloak-js](https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter) in Angular applications providing the following features:
+
+- A **Keycloak Service** which wraps the `keycloak-js` methods to be used in Angular, giving extra
+  functionalities to the original functions and adding new methods to make it easier to be consumed by
+  Angular applications.
+- Generic **AuthGuard implementation**, so you can customize your own AuthGuard logic inheriting the authentication logic and the roles load.
+- A **HttpClient interceptor** that adds the authorization header to all HttpClient requests.
+  It is also possible to disable this interceptor or exclude routes from having the authorization header.
+- This documentation also assists you to configure the keycloak in your Angular applications and with
+  the client setup in the admin console of your keycloak installation.
+
+## Setup
+
+In order to make sure Keycloak is initialized when your application is bootstrapped you will have to add an `APP_INITIALIZER` provider to your `AppModule`. This provider will call the `initializeKeycloak` factory function shown below which will set up the Keycloak service so that it can be used in your application.
+
+Use the code provided below as an example and implement it's functionality in your application. In this process ensure that the configuration you are providing matches that of your client as configured in Keycloak.
+
+```ts
+import { APP_INITIALIZER, NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { KeycloakAngularModule, KeycloakService } from 'keycloak-angular';
+import { AppRoutingModule } from './app-routing.module';
+import { AppComponent } from './app.component';
+
+function initializeKeycloak(keycloak: KeycloakService) {
+  return () =>
+    keycloak.init({
+      config: {
+        url: 'http://localhost:8080',
+        realm: 'your-realm',
+        clientId: 'your-client-id'
+      },
+      initOptions: {
+        onLoad: 'check-sso',
+        silentCheckSsoRedirectUri: window.location.origin + '/assets/silent-check-sso.html'
+      }
+    });
+}
+
+@NgModule({
+  declarations: [AppComponent],
+  imports: [AppRoutingModule, BrowserModule, KeycloakAngularModule],
+  providers: [
```

Comment:
The APP_INITIALIZER token is deprecated starting Angular 19, you can replace it with something like this :

`provideAppInitializer(
      () => {
        const keycloak=  inject(KeycloakService)
        keycloak.init({
          config: {
            url: 'http://localhost:8080',
            realm: 'your-realm',
            clientId: 'your-client-id'
          },
          initOptions: {
            onLoad: 'check-sso',
            silentCheckSsoRedirectUri: window.location.origin + '/assets/silent-check-sso.html'
          }
        })
      }
    )`